packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -124,31 +126,65 @@ async function pushToQueue(task, cb) {
  * @param  {Headers}  headers
  * @param  {String}   tmpFilename
  * @param  {Object}   httpOpts
+ * @param  {number}   attempt
  * @return {Promise<Object>}  Resolves with the [http Result Object]{@link https://nodejs.org/api/http.html#http_class_http_serverresponse}
  */
-const requestRemoteNode = (url, headers, tmpFilename, httpOpts) =>
+const requestRemoteNode = (url, headers, tmpFilename, httpOpts, attempt = 1) =>
   new Promise((resolve, reject) => {
-    const opts = Object.assign({}, { timeout: 30000, retries: 5 }, httpOpts)
+    let timeout
+
+    // Called if we stall for 30s without receiving any data
+    const handleTimeout = () => {
+      fsWriteStream.close()
+      fs.removeSync(tmpFilename)
+      if (attempt < RETRY_LIMIT) {
+        // Retry by calling ourself recursively
+        resolve(
+          requestRemoteNode(url, headers, tmpFilename, httpOpts, attempt + 1)
+        )
```

Comment:
should we try to re-implement exponential backoff before re-trying? or the code as-is work well enough?

Reply:
So far it's never needed to retry more than once. I think backoff would make sense if we were retrying after catching errors, but these stalls seem to be totally random and can be retried immeditely